pkg/router/template/router.go

```diff
@@ -29,6 +30,7 @@ import (
 )
 
 var log = logf.Logger.WithName("template")
+var noChildProcessesRegExp = regexp.MustCompile("[wait|waitid]: no child processes")
```

Comment:
Should be `(wait|waitid): no child processes` (or `wait(?:id): no child processes`).  Using brackets works by accident because `[wait|waitid]` matches the initial "t" in "t: no child processes" or the initial "d" in "d: no child processes" and the expression is not anchored.

Reply:
> Should be `(wait|waitid): no child processes` (or `wait(?:id)?: no child processes`). Using brackets works by accident because `[wait|waitid]` matches the initial "t" in "t: no child processes" or the initial "d" in "d: no child processes" and the expression is not anchored.

Right && thanks.

This is (now) all a hack. It doesn't reveal the root cause or explain why this failure is so sporadic.

Reply:
> Should be `(wait|waitid): no child processes` (or `wait(?:id)?: no child processes`). Using brackets works by accident because `[wait|waitid]` matches the initial "t" in "t: no child processes" or the initial "d" in "d: no child processes" and the expression is not anchored.

I only dropped in the additional "wait" as I got different errors locally compared to what I see reported by CI.